rpa-home/modules/ROOT/pages/automation-usecase-legacysystemdataextraction-extractfromscreen.adoc

```diff
@@ -0,0 +1,97 @@
+
+image:usecase-legacysystem-screeninteraction-mouseaction.png["The coordinates from the found pattern are used in the Relative To field of the Mouse Action Action Step."]
+
+The search field now has the focus, so you can just enter the order ID with a simulated keyboard. Instead of recording key strokes, you use the Activity Parameter `order_id` with the *Enter String* Action Step to copy the Order ID into the field:
+
+image::usecase-legacysystem-screeninteraction-enterstring.png[]
+
+You use *Mouse Action* again to start the search. To determine the position of the search button to click, you reuse the location of the `Order id` pattern to record the click position.
+
+image::usecase-legacysystem-screeninteraction-mouseaction-clickbutton.png[]
+
+## Check the Results
+
+You know that the search is successful when there is a result listed. As you can see on xref:automation-usecase-legacysystemdataextraction.adoc#trackingsystem-allentries-image[the screen], the tracking number should be listed in the column of the same name, so you search for that column first, using `Tracking No.` as the search pattern:
+
+image::usecase-legacysystem-screeninteraction-searchpattern-text.png[]
+
+Because it might be that the order has no corresponding tracking number in the system and you don't want to falsely return the initial value of the variable as tracking number, you check this case first using the *Managed* Action Step.
+
+In the *DoAction*, which works like a try element, you check if the result list is empty. As you can see on xref:automation-usecase-legacysystemdataextraction.adoc#trackingsystem-noentries-image[the screen], in case of no results, an up arrow is shown directly below the empty space where the results should have been.
+
+You use the *Search Pattern* Action Step to search for this arrow up symbol and limit the search area to the space directly below the column name.
+
+If the bot doesn't find the pattern- which in your case means that the system found a corresponding tracking number - *Search Pattern* returns an error. To continue with the workflow you need to force an OK state then, which you do in the *OnError* section.
+
+image::usecase-legacysystem-screeninteraction-searchpattern-relatedcoordinates.png[]
+
+If the legacy application returns the tracking number, the number of found patterns matching the arrow up symbol is zero. You use this information as a case differentiator in the *Select Case* Action Step.
+
+image::usecase-legacysystem-screeninteraction-selectcase.png["The Select Case Action Step uses Found Matched Pattern Count as Pin Variable Found Matched Pattern Count. The case compares the value to zero."]
+
+If there is a result, you set the gateway variable `is_orderid_found` to `true`.
+
+image::usecase-legacysystem-screeninteraction-selectcase-orderfoundtrue.png[]
```

Comment:
double-check images without alt text. 

If you intentionally want no alt text, you must include an empty quote set. Otherwise, the screen reader will read that file name.